### PR TITLE
tcl-fcgi: init at 0.5.0

### PR DESCRIPTION
**Pull request by nat-418**
###### Description of changes

Add Tcl package / library for interfacing with FastCGI.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tcl-fcgi</li>
  </ul>
</details>

File: pkgs/development/libraries/tcl-fcgi/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/${src.owner}/${src.repo}";
+    description = "Tcl interface for the FastCGI protocol";
+    license = licenses.bsd2;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ nat-418 ];
+  };
+}
+
+
```
Comment (Janik-Haag) [inline pkgs/development/libraries/tcl-fcgi/default.nix:32]: Please, remove the trailing line breaks
```suggestion
```

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
It would be nice if you change the commit message to `tcl-fcgi: init at 0.5.0`

**Review by Janik-Haag — APPROVED** [review]

**Comment by nat-418** [comment]
sorry @Janik-Haag for requesting you review again by accident

File: pkgs/development/libraries/tcl-fcgi/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, fetchFromGitHub, tcl, tclx }:
+
+tcl.mkTclDerivation rec {
+  pname = "tcl-fcgi";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "mpcjanssen";
+    repo = "tcl-fcgi";
+    rev = "HEAD";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tcl-fcgi/default.nix:10]: This must be pinned to a commit or better tag

File: pkgs/development/libraries/tcl-fcgi/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    tclx
+  ];
+
+  unpackPhase = ''
+    mkdir -p $out/lib/tcl-fcgi
+    cp -r $src/tcl-src/* $out/lib/tcl-fcgi/
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/${src.owner}/${src.repo}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tcl-fcgi/default.nix:24]:
```suggestion
    homepage = "https://github.com/mpcjanssen/tcl-fcgi";
```
I can't copy the substituted variant

File: pkgs/development/libraries/tcl-fcgi/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib, fetchFromGitHub, tcl, tclx }:
+
+tcl.mkTclDerivation rec {
+  pname = "tcl-fcgi";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "mpcjanssen";
+    repo = "tcl-fcgi";
+    rev = "62452db";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tcl-fcgi/default.nix:10]:
```suggestion
    rev = "62452dbf3177ba9458fbb42457834ca77bdf5a82";
```
we must use long hashes

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**